### ISSUE TITLE
[coroutine] Clang gives incorrect warnings

### ISSUE THREAD

**Issue by debashish-ghosh** [llvm/llvm-project#56768]
In the following code snippet:
```cpp
#include <coroutine>
struct coro {
  struct promise_type {
    coro get_return_object();
    constexpr auto initial_suspend() { return std::suspend_never{}; }
    constexpr auto final_suspend() noexcept { return std::suspend_always{}; }
    void unhandled_exception();
    void return_void();
  };
};
coro c(int i) {
  co_await (i = 0, std::suspend_always{});
}
```
clang gives a weird warning:
```console
warning: multiple unsequenced modifications to 'i' [-Wunsequenced]
  co_await (i = 1, std::suspend_always());
              ^
1 warning generated.
```

Another example can be found [here](https://godbolt.org/z/vMnc77Er6).

**Comment by ecatmur** [comment]
It looks like this is caused by https://reviews.llvm.org/D115187 https://github.com/llvm/llvm-project/commit/df2a4eae6b190806c0b96ef3312975e1c97dbda0

the fix will be to skip `CoroutineSuspendExpr` in SemaChecking/CheckUnsequencedOperations, like we do in AnalyzeImplicitConversions.

**Comment by r-barnes** [comment]
We observe what we think is the same issue when doing, eg,
```
co_await foo(attempts++)
```

**Comment by bcardosolopes** [comment]
A patch is up in https://reviews.llvm.org/D142077

**Comment by bcardosolopes** [comment]
Fixed in https://reviews.llvm.org/rG0b8daee028a87ab8a6f8fe54d2eb2d5b5c2babd4
